### ISSUE TITLE
[rush] Custom "watch" command -- only running on subset even with high -p, no log passthrough

### ISSUE THREAD

**Issue by natew** [microsoft/web-build-tools#532]
Hey, just playing with Rush and it seems pretty snazzy.

We use a big monorepo and have something in Lerna now that basically does `npm run build-watch` parallel in all packages. This is to run babel and watch for changes as you develop.

I wasn't sure the best way to do this in Rush. I added a custom command that seems to work, and then run it like: `rush watch -p 100 -v`.

Two things:

1. While it says "Registered" for all our packages (25) and says `Executing a maximum of 100 simultaneous processes...`, it only says `started` for 14 of them and doesn't seem to run beyond those 14. 
2. It doesn't output any logs from the corresponding processes, so if there's a syntax error in one it's hidden.

Perhaps I'm not doing things the Rush way here. Is there support for long-running development processes in some better way?

**Comment by natew** [comment]
One note, we the watch command doesn't rebuild anything it uses the babel-cli flag `--skip-initial-build`, which is helpful so you don't have to re-run builds all the time.

**Comment by HipsterZipster** [comment]
@nickpape-msft Thanks again for creating such an amazing project that's designed for enterprise. We are major users of Rush and have been promoting it across my org. 

Can you please provide a link to the pr or details on this "changed-projects-only" functionality? I searched in the repo and couldn't find it. More specifically, I'd really like to understand how this fits into the SDLC? Does this assume that devs check in the build artifacts for each project? If not, how else would a build server be able to proceed with building the downstream packages?

Thanks

**Comment by halfnibble** [comment]
Rush looks into each project in your mono repo and creates a dependency graph. It determines which projects have changed, and then builds everything that has changed and/or is affected by another package's change in the order necessary to propagate said changes. `rush build` does this. If you have 25 projects in your mono repo, then you would need a CPU with about 16 cores and at least 32 GB of RAM to run a rush command on every project at the same time because rush runs commands in separate child processes.

Basically, rush is not designed for this use case (according to my understanding). However, I want to look at some alternative ideas (maybe webpack's devServer can be configured to load 25 different projects?). If I can think of something, then I'll try to integrate it with rush.

**Comment by halfnibble** [comment]
I'm going to close this for now. If you have more ideas or questions, then send me an email.

**Comment by octogonz** [comment]
Agreed.  This issue is the same as https://github.com/microsoft/web-build-tools/issues/1202, which includes a somewhat complete current design proposal for this feature.

@halfnibble
